pkg/cloud/services/loadbalancer/loadbalancer.go

```diff
@@ -287,7 +287,7 @@ func (s *Service) getOrUpdateAllowedCIDRS(openStackCluster *infrav1.OpenStackClu
 
 		listener, err := s.loadbalancerClient.UpdateListener(listener.ID, listenerUpdateOpts)
 		if err != nil {
-			record.Warnf(openStackCluster, "FailedUpdateListener", "Failed to update listener %s: %v", listener.Name, err)
+			record.Warnf(openStackCluster, "FailedUpdateListener", "Failed to update listener: %v", err)
```

Comment:
instead of using Name, could we just use ID? So the user has still a way to find which LB wasn't update.

Reply:
The problem here is that listener is nul due to https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/4fa8cd15475ccf81aaccf37b348611a221c502e4/pkg/clients/loadbalancer.go#L126

I can verify this but I think this will result in the same problem

Reply:
listenerName is included when this function is called. I change to code to use that variable instead to avoid adding any variables just used for a potential error message